front/lib/swr.ts

```diff
@@ -1028,6 +1044,78 @@ export function useWorkspaceEnterpriseConnection({
   };
 }
 
+interface UseDataSourceKey {
+  workspaceid: string;
+  dataSourceName: string;
+  internalIds: string[];
+}
+
+export function useDataSourceNodes(
+  key: UseDataSourceKey,
+  options?: SWRConfiguration<{
+    contentNodes: ContentNode[];
+    parentsById: Record<string, Set<string>>;
+  }>
+) {
+  const contentNodesFetcher: Fetcher<
+    { contentNodes: ContentNode[]; parentsById: Record<string, Set<string>> },
+    string
+  > = async (key: string) => {
+    const { workspaceSid, dataSourceName, internalIds } = JSON.parse(key);
+    if (internalIds.length === 0 || !dataSourceName) {
```

Comment:
Looking at the type `UseDataSourceKey` `dataSourceName` is always present, right? I guess we don't need this check?